Wikipedia/UI-V5/WMFArticleListDataSource.h

```diff
@@ -17,9 +18,13 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (MWKHistoryDiscoveryMethod)discoveryMethod;
 
+- (void)setSavedPageList:(MWKSavedPageList*)savedPageList;
```

Comment:
> Any reason for this to be mutable as opposed to initing with it?

Can't be passed as an initialization parameter because:

1. the data sources have their own designated initializers 
2. the article list doesn't create them

Maybe we can make `initWithSavedPageList:` the designated initializer of the [shared base class](https://phabricator.wikimedia.org/T111571), until then this was the easiest way.

We needed a way to propagate the saved page list down the object graph